src/rust/src/functions/range.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{datatypes::PlRDataType, prelude::*, PlRExpr, PlRSeries, RPolarsErr};
+use polars::lazy::dsl;
+use savvy::{savvy, NumericScalar, Result};
+
+#[savvy]
+pub fn int_range(
+    start: &PlRExpr,
+    end: &PlRExpr,
+    step: NumericScalar,
+    dtype: &str,
+) -> Result<PlRExpr> {
+    let start = start.inner.clone();
+    let end = end.inner.clone();
+    let dtype = PlRDataType::try_from(dtype)?.dt;
+    let step = <Wrap<i64>>::try_from(step)?.0;
+    Ok(dsl::int_range(start, end, step, dtype).into())
+}
+
+// TODO-REWRITE: unsure of the replacement for Bound
+// #[savvy]
+// pub fn eager_int_range(
+//     lower: &Bound<'_, PyAny>,
+//     upper: &Bound<'_, PyAny>,
+//     step: &Bound<'_, PyAny>,
+//     dtype: &str,
+// ) -> Result<PlRSeries> {
+//     let dtype = PlRDataType::try_from(dtype)?.dt;
+//     if !dtype.is_integer() {
+//         return Err(RPolarsErr::from(
+//             polars_err!(ComputeError: "non-integer `dtype` passed to `int_range`: {:?}", dtype),
+//         )
+//         .into());
+//     }
+
+//     let ret = with_match_physical_integer_polars_type!(dtype, |$T| {
+//         let start_v: <$T as PolarsNumericType>::Native = lower.extract()?;
+//         let end_v: <$T as PolarsNumericType>::Native = upper.extract()?;
+//         let step: i64 = step.extract()?;
+//         new_int_range::<$T>(start_v, end_v, step, PlSmallStr::from_static("literal"))
+//     });
+
+//     let s = ret.map_err(RPolarsErr::from)?;
+//     Ok(s.into())
+// }
+
+#[savvy]
+pub fn int_ranges(start: &PlRExpr, end: &PlRExpr, step: &PlRExpr, dtype: &str) -> Result<PlRExpr> {
```

Comment:
Why is dtype `&str`?
it seems to be different from Python Polars.
https://github.com/pola-rs/polars/blob/01a4e067f50fd5f8825d4b25402336310e125d8d/crates/polars-python/src/functions/range.rs#L45-L66